src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+    private static final String MESSAGE_WELCOME = "Hello! I'm FredBot.\nWhat can I do for you?";
+
+    private static final String PREFIX_BY = "/by";
+    private static final String PREFIX_FROM = "/from";
+    private static final String PREFIX_TO = "/to";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        initTaskList();
+        showWelcomeMessage();
+        while (true) {
+            String command = readUserInput();
+            executeCommand(command);
+        }
+    }
+
+    private static void initTaskList() {
+        allTasks = new Task[TASK_CAPACITY];
+        count = 0;
+    }
+
+    private static void showWelcomeMessage() {
+        System.out.println(MESSAGE_WELCOME);
+    }
+
+    private static String readUserInput() {
+        return SCANNER.nextLine();
+    }
+
+    private static void executeCommand(String input) {
+        final String[] commandWordAndArgs = splitCommandWordAndArgs(input);
+        final String commandWord = commandWordAndArgs[0];
+        final String commandArgs = commandWordAndArgs[1];
+
+        switch (commandWord) {
+            case COMMAND_TODO:
+                executeAddTodo(commandArgs);
+                break;
+            case COMMAND_DEADLINE:
+                executeAddDeadline(commandArgs);
+                break;
+            case COMMAND_EVENT:
+                executeAddEvent(commandArgs);
+                break;
+            case COMMAND_LIST:
+                executeList();
+                break;
+            case COMMAND_MARK:
+                executeMark(commandArgs);
+                break;
+            case COMMAND_UNMARK:
+                executeUnmark(commandArgs);
+                break;
+            case COMMAND_EXIT:
+                executeExit();
+                break;
+            default:
+                executeError();
+                break;
+        }
+    }
+
+    private static String[] splitCommandWordAndArgs(String input) {
+        final String[] split = input.split(" ", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""};
+    }
+
+    private static void executeAddTodo(String input) {
+        allTasks[count] = new Todo(input);
+        echoTask();
+    }
+
+    private static void executeAddDeadline(String input) {
+        String[] split = splitDeadlineAndDate(input);
+        allTasks[count] = new Deadline(split[0], split[1]);
+        echoTask();
+    }
+
+    private static String[] splitDeadlineAndDate(String input) {
+        final int indexOfByPrefix = input.indexOf(PREFIX_BY);
```

Comment:
Perhaps remove the `final` keyword? As the scope of this variable `indexOfByPrefix` is only within the method; tagging it with a final might mislead other readers that `indexOfByPrefix` can be modified outside the method without the `final`